lib/preview/previewUtils.js

```diff
@@ -53,16 +41,30 @@ const insertAtEndOfBody = (html, script) => {
 const addRefreshScript = (html) => {
   const refreshScript = `
   <script>
-  setInterval(() => {
-    const res = fetch('/meta')
-      .then(async (res) => {
-        const hsServerState = await res.json();
-        if (hsServerState["REMOTE_FS_IS_DIRTY"]) {
-          location.reload();
-        }
-      })
-      .catch(err => console.log(err));
-  }, 1000);
+  (() => {
+    const NORMAL_WAIT_MS = 1000;
+    const BACKOFF_RATIO = 2;
+    let nextWait = NORMAL_WAIT_MS;
+    let timeWaited = 0;
+
+    setInterval(() => {
+      timeWaited += NORMAL_WAIT_MS;
+      if (timeWaited !== nextWait) return;
+      timeWaited = 0;
+      const res = fetch(window.location.origin + '/meta')
+        .then(async (res) => {
+          const hsServerState = await res.json();
+          if (hsServerState["REMOTE_FS_IS_DIRTY"]) {
+            location.reload();
+          }
+          nextWait = NORMAL_WAIT_MS;
+        })
+        .catch(err => {
+          nextWait *= BACKOFF_RATIO;
+          console.log('Disconnected from local server... (retrying in ' + nextWait / 1000 + 's)');
+        });
+    }, NORMAL_WAIT_MS);
+  })();
```

Comment:
Exponential backoff if connection to local server is servered as middle ground between giving up and continuously polling

Reply:
Might be worth adding a maximum here (maybe when nextWait = 1 min or something)